### PR TITLE
test(docker): SSL path adjustments for linux alpine 3.21

### PR DESCRIPTION
**Pull request by jkomyno**
This PR adds the following Docker tests:
- `alpine-3.21-amd64-openssl-3.0.x`
- `alpine-3.21-arm64-openssl-3.0.x`

It also modifies:
- `alpine-latest-amd64-openssl-3.0.x`
- `alpine-latest-arm64-openssl-3.0.x`

### CODE REVIEW

**Comment by jkomyno** [comment]
Note: https://github.com/prisma/ecosystem-tests/actions/runs/12294941892/job/34310918073?pr=5703 is still failing.

File: docker/alpine-3.21-amd64-openssl-3.0.x/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+FROM node:22-alpine3.21
+
+WORKDIR /usr/src/app
+COPY --from=app . ./
+COPY --from=utils . ./
+
+RUN apk update
+
+# Only OpenSSL 3 is expected to be on this system
+RUN if [ ! -e /usr/lib/libssl.so.3 ]; then echo 'OpenSSL 3 not found in /lib/libssl.so.3'; exit 1; fi
```
Comment (aqrln) [inline docker/alpine-3.21-amd64-openssl-3.0.x/Dockerfile]:
```suggestion
RUN if [ ! -e /usr/lib/libssl.so.3 ]; then echo 'OpenSSL 3 not found in /usr/lib/libssl.so.3'; exit 1; fi
```

File: docker/alpine-3.21-amd64-openssl-3.0.x/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+RUN if [ -e /usr/lib/libssl.so.1.1 ]; then echo 'OpenSSL 1.1 must not be installed (found in /lib/libssl.so.1.1)'; exit 1; fi
```
Comment (aqrln) [inline docker/alpine-3.21-amd64-openssl-3.0.x/Dockerfile]:
```suggestion
RUN if [ -e /usr/lib/libssl.so.1.1 ]; then echo 'OpenSSL 1.1 must not be installed (found in /usr/lib/libssl.so.1.1)'; exit 1; fi
```

File: docker/alpine-3.21-arm64-openssl-3.0.x/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+WORKDIR /usr/src/app
+COPY --from=app . ./
+COPY --from=utils . ./
+
+RUN apk update
+
+# Only OpenSSL 3 is expected to be on this system
+RUN if [ ! -e /usr/lib/libssl.so.3 ]; then echo 'OpenSSL 3 not found in /lib/libssl.so.3'; exit 1; fi
+RUN if [ -e /usr/lib/libssl.so.1.1 ]; then echo 'OpenSSL 1.1 must not be installed (found in /lib/libssl.so.1.1)'; exit 1; fi
```
Comment (aqrln) [inline docker/alpine-3.21-arm64-openssl-3.0.x/Dockerfile]:
```suggestion
RUN if [ ! -e /usr/lib/libssl.so.3 ]; then echo 'OpenSSL 3 not found in /usr/lib/libssl.so.3'; exit 1; fi
RUN if [ -e /usr/lib/libssl.so.1.1 ]; then echo 'OpenSSL 1.1 must not be installed (found in /usr/lib/libssl.so.1.1)'; exit 1; fi
```

**Pull request merged**